osu.Game.Rulesets.Osu.Tests/Mods/TestSceneOsuModRelax.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using NUnit.Framework;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Osu.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Replays;
+using osu.Game.Rulesets.Osu.UI;
+using osu.Game.Scoring;
+using osu.Game.Tests.Visual;
+
+namespace osu.Game.Rulesets.Osu.Tests.Mods
+{
+    public class TestSceneOsuModRelax : OsuModTestScene
+    {
+        [Test]
+        public void Test()
+        {
+            var lastObject = new HitCircle()
+            {
+                Position = OsuPlayfield.BASE_SIZE / 2,
+                StartTime = 2550
+            };
+            CreateModTest(new ModTestData
+            {
+                Autoplay = false,
+                Mods =
+                    new Mod[]
+                    {
+                        new OsuModRelax(),
+                        new OsuModAutopilot()
+                    },
+                Beatmap = new Beatmap
+                {
+                    HitObjects =
+                    {
+                        new HitCircle()
+                        {
+                            Position = OsuPlayfield.BASE_SIZE / 2,
+                            StartTime = 2500
+                        },
+                        lastObject
+                    }
+                },
+                PassCondition = () => Player.ScoreProcessor.JudgedHits == 2
+            });
+            Score score = null;
+
+            AddUntilStep("wait for beatmap end", () => Player.GameplayClockContainer.GameplayClock.CurrentTime >= lastObject.GetEndTime());
+            AddStep("get score", () => score = ((ScoreAccessiblePlayer)Player).Score);
+            AddAssert("no key was held", () => score.Replay.Frames.Cast<OsuReplayFrame>().TakeLast(2).FirstOrDefault().Actions.Count == 0);
```

Comment:
No idea why this fails, can pass locally
![image](https://user-images.githubusercontent.com/50285552/126114738-daf3207d-4189-4b60-8fdd-ead824972e1e.png)

Reply:
Run from nunit locally. You will reproduce the failure.

Reply:
I don't understand. Why this issue must be seen in replays, and the key counter during gameplay not showing this issue?
And asseting for the last frame has 0 actions all failed

Reply:
Please learn to use your debugger, and unit testing:

https://user-images.githubusercontent.com/191335/126132756-ee835e48-c728-4c1b-9a78-812f51b72c10.mp4

I haven't looked into why this is the case, but you should be able to figure things out from the above, right?